pydatastructs/trees/_backend/cpp/BinarySearchTree.hpp

```
@@ -0,0 +1,715 @@
+                bt->size = bt->size + 1;
+                flag = false;
+            }
+            prev_node = reinterpret_cast<TreeNode*>(bt->tree->_one_dimensional_array->_data[PyLong_AsLong(walk)])->right;
+            walk = reinterpret_cast<TreeNode*>(bt->tree->_one_dimensional_array->_data[PyLong_AsLong(walk)])->right;
+        }
+        else {
+            if (reinterpret_cast<TreeNode*>(bt->tree->_one_dimensional_array->_data[PyLong_AsLong(walk)])->left == Py_None) {
+                new_node->parent = prev_node;
+                ArrayForTrees_append(bt->tree, Py_BuildValue( "[O]", reinterpret_cast<PyObject*>(new_node)) );
+                reinterpret_cast<TreeNode*>(bt->tree->_one_dimensional_array->_data[PyLong_AsLong(walk)])->left = PyLong_FromLong(bt->size);
+                bt->size = bt->size + 1;
+                flag = false;
+            }
+            prev_node = reinterpret_cast<TreeNode*>(bt->tree->_one_dimensional_array->_data[PyLong_AsLong(walk)])->left;
+            walk = reinterpret_cast<TreeNode*>(bt->tree->_one_dimensional_array->_data[PyLong_AsLong(walk)])->left;
+        }
+    }
+    BinarySearchTree__update_size(self, Py_BuildValue("(O)",walk));
+    Py_RETURN_NONE;
+}
+
+static PyObject* BinarySearchTree_delete(BinarySearchTree* self, PyObject *args, PyObject *kwds) {
+    Py_INCREF(Py_None);
+    PyObject* key = Py_None;
+    Py_INCREF(Py_None);
+    PyObject* balancing_info = Py_None;
+    static char* keywords[] = {"key","balancing_info", NULL};
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|O", keywords, &key, &balancing_info)) {
+        return NULL;
+    }
+    PyObject* kwd_parent = PyDict_New();
+    PyDict_SetItemString(kwd_parent, "parent", PyLong_FromLong(1));
+    PyObject* tup = BinarySearchTree_search(self, Py_BuildValue("(O)",key), kwd_parent);
+    PyObject* walk = PyTuple_GetItem(tup, 0);
+    PyObject* parent = PyTuple_GetItem(tup, 1);
+    Py_INCREF(Py_None);
+    PyObject* a = Py_None;
```

May be this never got updated?

---

Try setting it to `PyObject* a = NULL`; That should raise an error, if it never gets updated.

---

The value of a is set properly. I checked by printing the value of a (after the statement where it should be updated).

---

Check all the points where this function returns None.